builtin/game/falling.lua

```diff
@@ -56,78 +48,91 @@ core.register_entity(":__builtin:falling_node", {
 		if not vector.equals(acceleration, {x = 0, y = -10, z = 0}) then
 			self.object:set_acceleration({x = 0, y = -10, z = 0})
 		end
-		-- Turn to actual node when colliding with ground, or continue to move
+
 		local pos = self.object:get_pos()
 		-- Position of bottom center point
-		local bcp = {x = pos.x, y = pos.y - 0.7, z = pos.z}
-		-- 'bcn' is nil for unloaded nodes
-		local bcn = core.get_node_or_nil(bcp)
+		local below_pos = {x = pos.x, y = pos.y - 0.7, z = pos.z}
+		-- Avoid bugs caused by an unloaded node below
+		local below_node = core.get_node(below_pos)
 		-- Delete on contact with ignore at world edges
-		if bcn and bcn.name == "ignore" then
+		if below_node.name == "ignore" then
 			self.object:remove()
 			return
 		end
-		local bcd = bcn and core.registered_nodes[bcn.name]
-		if bcn and
-				(not bcd or bcd.walkable or
-				(core.get_item_group(self.node.name, "float") ~= 0 and
-				bcd.liquidtype ~= "none")) then
-			if bcd and bcd.leveled and
-					bcn.name == self.node.name then
-				local addlevel = self.node.level
-				if not addlevel or addlevel <= 0 then
-					addlevel = bcd.leveled
-				end
-				if core.add_node_level(bcp, addlevel) == 0 then
-					self.object:remove()
-					return
-				end
-			elseif bcd and bcd.buildable_to and
-					(core.get_item_group(self.node.name, "float") == 0 or
-					bcd.liquidtype == "none") then
-				core.remove_node(bcp)
+
+		local below_nodef = core.registered_nodes[below_node.name]
+		-- Is it a level node we can add to?
+		if below_nodef
+		and below_nodef.leveled
```

Comment:
Needs two additional indents, preferably you could put these checks into the same line